repo/fsrepo/migrations/common/testing_helpers.go

```diff
@@ -0,0 +1,233 @@
+
+	tempDir := t.TempDir()
+
+	// Write version file
+	versionPath := filepath.Join(tempDir, "version")
+	err := os.WriteFile(versionPath, []byte(fmt.Sprintf("%d", version)), 0644)
+	if err != nil {
+		t.Fatalf("failed to write version file: %v", err)
+	}
+
+	// Write config file
+	configPath := filepath.Join(tempDir, "config")
+	configData, err := json.MarshalIndent(config, "", "  ")
+	if err != nil {
+		t.Fatalf("failed to marshal config: %v", err)
+	}
+	err = os.WriteFile(configPath, configData, 0644)
+	if err != nil {
+		t.Fatalf("failed to write config file: %v", err)
+	}
+
+	return tempDir
+}
+
+// AssertMigrationSuccess runs a full migration and checks that it succeeds
+func AssertMigrationSuccess(t *testing.T, migration Migration, fromVersion, toVersion int, inputConfig map[string]any) map[string]any {
+	t.Helper()
+
+	// Create test repo
+	repoPath := CreateTestRepo(t, fromVersion, inputConfig)
+
+	// Run migration
+	opts := Options{
+		Path:    repoPath,
+		Verbose: false,
+	}
+
+	err := migration.Apply(opts)
+	if err != nil {
+		t.Fatalf("migration failed: %v", err)
+	}
+
+	// Check version was updated
+	versionBytes, err := os.ReadFile(filepath.Join(repoPath, "version"))
+	if err != nil {
+		t.Fatalf("failed to read version file: %v", err)
+	}
+	actualVersion := string(versionBytes)
+	if actualVersion != fmt.Sprintf("%d", toVersion) {
+		t.Errorf("expected version %d, got %s", toVersion, actualVersion)
+	}
+
+	// Read and return the migrated config
+	configBytes, err := os.ReadFile(filepath.Join(repoPath, "config"))
+	if err != nil {
+		t.Fatalf("failed to read config file: %v", err)
+	}
+
+	var result map[string]any
+	err = json.Unmarshal(configBytes, &result)
+	if err != nil {
+		t.Fatalf("failed to unmarshal config: %v", err)
+	}
+
+	return result
+}
+
+// AssertMigrationReversible checks that a migration can be reverted
+func AssertMigrationReversible(t *testing.T, migration Migration, fromVersion, toVersion int, inputConfig map[string]any) {
+	t.Helper()
+
+	// Create test repo at target version
+	repoPath := CreateTestRepo(t, toVersion, inputConfig)
+
+	// Create backup file (simulating a previous migration)
+	backupPath := filepath.Join(repoPath, fmt.Sprintf("config.%d-to-%d.bak", fromVersion, toVersion))
+	originalConfig, _ := json.MarshalIndent(inputConfig, "", "  ")
+	err := os.WriteFile(backupPath, originalConfig, 0644)
+	if err != nil {
+		t.Fatalf("failed to write backup file: %v", err)
+	}
+
+	// Run revert
+	opts := Options{
+		Path:    repoPath,
+		Verbose: false,
+	}
+
+	err = migration.Revert(opts)
+	if err != nil {
+		t.Fatalf("revert failed: %v", err)
+	}
+
+	// Check version was reverted
+	versionBytes, err := os.ReadFile(filepath.Join(repoPath, "version"))
+	if err != nil {
+		t.Fatalf("failed to read version file: %v", err)
+	}
+	actualVersion := string(versionBytes)
+	if actualVersion != fmt.Sprintf("%d", fromVersion) {
+		t.Errorf("expected version %d after revert, got %s", fromVersion, actualVersion)
+	}
+}
```

Comment:
Maybe add check that some config item has been reverted to its previous value.

Reply:
Good idea, added in 99a1fc209333c94756bfd539c6f3199b8da1dbb1